arcgis-ios-sdk-samples/Display information/Display grid/DisplayGridViewController.swift

```diff
@@ -41,34 +39,47 @@ class DisplayGridViewController: UIViewController, UIAdaptivePresentationControl
         mapView.grid = AGSLatitudeLongitudeGrid()
     }
     
-    //MARK: - UIAdaptivePresentationControllerDelegate
+    var settingsViewController: DisplayGridSettingsViewController?
     
-    func adaptivePresentationStyle(for controller: UIPresentationController, traitCollection: UITraitCollection) -> UIModalPresentationStyle {
-        //
-        // For popover or non modal presentation
-        return UIModalPresentationStyle.none
+    func makeSettingsViewController() -> DisplayGridSettingsViewController {
+        guard let viewController = storyboard?.instantiateViewController(withIdentifier: "DisplayGridSettingsViewController") as? DisplayGridSettingsViewController else {
+            fatalError()
```

Comment:
fatalError seems kind of abrupt, is there a way that execution could continue or does a storyboard problem mean that we are in an unstable state?  I didn't find any use of fatalError in the app so far; if the philosophy is to avoid stopping execution, I'm hesitant to introduce that pattern?

Reply:
@pgruenler In effect we use `fatalError()` all throughout the app; it just shows up as a force unwrap (`!`). I was just reading [a post on the Swift forums](https://forums.swift.org/t/se-0217-the-unwrap-or-die-operator/14107/222) that describes this practice as "a way of more thoughtfully crashing on nil with an actionable error message." 

That being said, I didn't include an actionable message. I could have, but at least using `fatalError()` communicates that the author considered the `nil` case and deemed it nigh impossible. With a force unwrap, the author may have added it out of convenience, not considering the consequences. Does that make sense?

Reply:
@philium, I see what you mean about `fatalError()` being a step up from forced unwrapping.  However, I would be a proponent of, as much as possible, not using either in our samples.  

@dg0yal, do you have any thoughts on this?  The code looks fine but to me it's a new pattern for our samples.  There are definitely two sides to the use of `fatalError()` and I fall more on the avoid it as much as possible side.